File: task-manager-api/src/middlewares/security.py
"""Security headers.

`flask-talisman` cobriria isto, mas é dependência nova — a regra é só adicionar
dependência quando o finding não puder ser resolvido sem ela.
"""


def register(app):
    @app.after_request
    def _headers(response):
        response.headers.setdefault("X-Content-Type-Options", "nosniff")
        response.headers.setdefault("X-Frame-Options", "DENY")
        response.headers.setdefault("Referrer-Policy", "no-referrer")
        return response

File: ecommerce-api-legacy/src/middlewares/security.js
'use strict';

/**
 * Security headers.
 *
 * `helmet` cobriria isto, mas é dependência nova — a regra da skill é só
 * adicionar dependência quando o finding não puder ser resolvido sem ela, e
 * quatro headers não justificam. Registrado antes das rotas.
 */
function securityHeaders(req, res, next) {
  res.setHeader('X-Content-Type-Options', 'nosniff');
  res.setHeader('X-Frame-Options', 'DENY');
  res.setHeader('Referrer-Policy', 'no-referrer');
  res.setHeader('X-Permitted-Cross-Domain-Policies', 'none');
  res.removeHeader('X-Powered-By');
  next();
}

module.exports = { securityHeaders };
